src/client/options/mod.rs

```diff
@@ -995,10 +1094,21 @@ impl ClientOptions {
         uri: impl AsRef<str>,
         resolver_config: Option<ResolverConfig>,
     ) -> Result<Self> {
-        let parser = ClientOptionsParser::parse(uri.as_ref())?;
-        let srv = parser.srv;
-        let auth_source_present = parser.auth_source.is_some();
-        let mut options: Self = parser.into();
+        Self::resolve(ConnectionString::parse(uri)?, resolver_config).await
+    }
+
+    /// Creates a `ClientOptions` from the given `ConnectionString`.
+    ///
+    /// In the case that "mongodb+srv" is used, SRV and TXT record lookups will be done using the
+    /// provided `ResolverConfig` as part of this method.
+    pub async fn resolve(
```

Comment:
can we indicate in this name somehow that the connection string is being resolved to `ClientOptions`? it might be slightly unreadable in user code to determine what the output of this method is if someone isn't super familiar with the interaction between the two types

Reply:
Good thought; renamed this to `ClientOptions::resolve_from`, which I think makes this more clear.